packages/permission-controller/src/PermissionController.ts

```diff
@@ -1773,17 +1797,10 @@ export class PermissionController<
         ControllerPermissionSpecification,
         ControllerCaveatSpecification
       >;
-      let performCaveatValidation = true;
-
       if (specification.factory) {
         permission = specification.factory(permissionOptions, requestData);
       } else {
         permission = constructPermission(permissionOptions);
-
-        // We do not need to validate caveats in this case, because the plain
-        // permission constructor function does not modify the caveats, which
-        // were already validated by `constructCaveats` above.
-        performCaveatValidation = false;
```

Comment:
We're all trying to find the guy who introduced this futile and brittle optimization.

Reply:
![image](https://github.com/user-attachments/assets/f054650f-41e8-4378-b94a-d17c32a36fff)